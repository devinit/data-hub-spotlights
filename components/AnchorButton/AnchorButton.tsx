import React, { FunctionComponent } from 'react';

type AnchorButtonProps = React.DetailedHTMLProps<React.AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>;

const AnchorButton: FunctionComponent<AnchorButtonProps> = ({ children, ...props }) => (
  <a {...props}>
    <style jsx>{`
      cursor: pointer;
    `}</style>
    {children}
  </a>
);

AnchorButton.defaultProps = { className: 'button' };

export { AnchorButton };
