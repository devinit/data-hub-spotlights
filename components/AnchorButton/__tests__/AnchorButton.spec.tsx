/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { AnchorButton } from '../AnchorButton';

describe('AnchorButton', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<AnchorButton>Caption</AnchorButton>).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
