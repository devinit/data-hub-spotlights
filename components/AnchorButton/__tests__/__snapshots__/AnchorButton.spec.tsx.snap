// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnchorButton renders correctly 1`] = `
<a
  className="button"
>
  <style
    jsx={true}
  >
    
      cursor: pointer;
    
  </style>
  Caption
</a>
`;
